tests/test_load.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+import re
+
+
+@pytest.mark.parametrize("invalid_layout,error_pos", [
+    ('(', 1),
+    ('()', 1),
+    ('foo baar', 0),
+    ('(foo baar', 1),
+    ('((clients max:0 ))', 1),
+    ('(clients)', 8),
+    ('(clients )', 9),
+    ('(split max:0.5:1)', 7),
+    ('(split horizontal:0.05:1)', 7),
+    ('(split horizontal:0.95:1)', 7),
+    ('(split horizontal:x:1)', 7),
+    ('(split horizontal:0.5:x)', 7),
+    ('(split horizontal:0.5:-1)', 7),
+    ('(split horizontal:0.5:2)', 7),
+    ('(split horizontal:0.3)', 7),
+    ('(split horizontal:0.3:0:0)', 7),
+    ('(split  horizonta:0.5:0 )', 8),
+    ('(clients max )', 9),
+    ('(clients max:0:0 )', 9),
+    ('(clients ma:0 )', 9),
+    ('(clients max:-1 )', 9),
+    ('(clients grid:0 asdf )', 16),
+    ('(clients grid:0 0xx0)', 16),
+    ('(clients grid:0 09)', 16),
+    ('(clients grid:0 0x)', 16),
+    ('(clients grid:0 x)', 16),
+    ('(split horizontal:0.5:0 x)', 24),
+    ('(split horizontal:0.5:0 (split horizontal:0.5:1', 47),
+    ('(split horizontal:0.5:0 (split horizontal:0.5:1 ', 48),
+    ('(split horizontal:0.5:0 (split horizontal:0.5:1 )', 49),
+    ('(split horizontal:0.5:0 (split horizontal:0.5:1 )))', 50),
+    ('(split horizontal:0.5:0 (clients max:1', 38),
+])
+def test_syntax_errors_position(hlwm, invalid_layout, error_pos):
+    c = hlwm.call_xfail(['load', invalid_layout])
+    c.expect_stderr(r'^load: Syntax error at {}: '.format(error_pos))
+
+
+def is_subseq(x, y):
+    """Checks if x is a subsequence (not substring) of y."""
+    # from https://stackoverflow.com/a/24017747/4400896
+    it = iter(y)
+    return all(c in it for c in x)
+
+
+@pytest.mark.parametrize("layout", [
+    "(clients max:0)",
+    "(clients grid:0)",
+    " (  clients   vertical:0  )",
+    "(split horizontal:0.3:0)",
+    "(split vertical:0.3:0 (clients horizontal:0))",
+    "(split vertical:0.3:0 (split vertical:0.4:1))",
+])
+@pytest.mark.parametrize('num_splits_before', [0, 1, 2])
+def test_valid_layout_syntax_partial_layouts(hlwm, layout, num_splits_before):
+    for i in range(0, num_splits_before):
+        hlwm.call('split explode')
+
+    hlwm.call(['load', layout])
+
+    assert is_subseq(layout.replace(' ', ''), hlwm.call('dump').stdout)
```

Comment:
Oof. I'm having a hard time understanding this.

Are you using is_subseq here only because the difference between a load and a subsequent dump is that the dump contains the `(clients ...)` components?

Reply:
Yes. The output of contains the full tree but in `layout` we didn't specify the tree in its entirety.

Reply:
I've added a comment to make this clearer in the test case. You're right that the test case as such is very hard to read.

Reply:
How would you feel about dealing with this by running the Python-equivalent of `s/\(clients [^)]+\)//g` on the `dump` output and then doing a regular string comparison? Would that work?

Reply:
No this would not work for e.g. `layout =  "(split vertical:0.3:0 (clients horizontal:0))"`. And if I did the replacement also in `layout` then I would test too little. Furthermore, the additional branches can also be `(split …)`-branches (this is the case for `layout = "(split horizontal:0.3:0)"` and `num_splits = 2`).

Reply:
Okay, got it (I think)!

Reply:
Sorry, for the lack of descriptive comments in the test case.